cpp/include/Ice/SSL/ClientAuthenticationOptions.h

```diff
@@ -98,326 +64,233 @@ namespace Ice::SSL
-         *
-         * Example of setting serverCertificateValidationCallback:
-         * ```cpp
-         * auto initData = Ice::InitializationData {
-         *   ...
-         *   .clientAuthenticationOptions = ClientAuthenticationOptions{
-         *     .serverCertificateValidationCallback =
-         *       [](SecTrustRef trust, const ConnectionInfoPtr& info)
-         *       {
-         *          ...
-         *          return SecTrustEvaluateWithError(trust, nullptr);
-         *       }
-         * });
-         * auto communicator = Ice::initialize(initData);
-         * ```
-         *
-         * @param context A CtxtHandle representing the security context associated with the current connection. This
-         * context contains security data relevant for validation, such as the client's certificate chain and cipher
-         * suite.
-         * @param info The ConnectionInfoPtr object that provides additional connection-related data which might
-         * be relevant for contextual certificate validation.
+         * A callback for validating the server certificate chain. If the verification callback returns false, the
+         * connection will be aborted with an Ice::SecurityException.
+         *
+         * When this callback is set, it replaces the default validation callback and must perform all necessary
+         * validation steps.
+         *
+         * Example of setting `serverCertificateValidationCallback`:
+         * @snippet Ice/SSL/SchannelClientAuthenticationOptions.cpp serverCertificateValidationCallback
+         *
+         * @param context An opaque object representing the security context associated with the current connection.
+         * This context contains security data relevant for validation, such as the server's certificate chain and
+         * cipher suite.
+         * @param info The connection info object that provides additional connection-related data. The `ConnectionInfo`
+         * type is an alias for the platform-specific connection info class.
          * @return true if the certificate chain is valid and the connection should proceed; false if the certificate
          * chain is invalid and the connection should be aborted.
          * @throws Ice::SecurityException if the certificate chain is invalid and the connection should be aborted.
-         *
-         * [See
-         * SecTrustEvaluateWithError](https://developer.apple.com/documentation/security/2980705-sectrustevaluatewitherror?language=objc)
+         * @see SSL::OpenSSLConnectionInfo
+         * @see SSL::SecureTransportConnectionInfo
+         * @see SSL::SchannelConnectionInfo
          */
         std::function<bool(CtxtHandle context, const ConnectionInfoPtr& info)> serverCertificateValidationCallback;
     };
-    // Alias for portable code
+
+    /// \cond INTERNAL
+    // Alias for the platform-specific implementation of ClientAuthenticationOptions on Windows.
```

Comment:
We need to keep this internal in the Doxygen documentation. Otherwise, the documentation will only show the first definition, which could be confusing.